Runtime/Components/ConeRayAnglesEstimator/AveragedConeRaySegmentComputation.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using ubco.ovilab.HPUI.Interaction;
+using UnityEngine;
+
+namespace ubco.ovilab.HPUI
+{
+    /// <summary>
+    /// Computes the cone ray angles by averaging accords frames.
+    /// </summary>
+    [Serializable]
+    public class AveragedConeRaySegmentComputation : IConeRaySegmentComputation
+    {
+        [SerializeField, Range(0.01f, 1f)]
+        [Tooltip("The percentage of frames in the gesture that a ray should have been used to qualify for the final cone")]
+        private float minRayInteractionsThreshold = 0.2f;
+
+        List<HPUIInteractorRayAngle> IConeRaySegmentComputation.EstimateConeAnglesForSegment(HPUIInteractorConeRayAngleSegment segment, IEnumerable<ConeRayComputationDataRecord> interactionRecords)
+        {
+            Dictionary<(float, float), float> averageRayDistance = new();
+            // For each interaction, get the frame with the shortest distance
+            foreach (ConeRayComputationDataRecord interactionRecord in interactionRecords)
+            {
+                if (interactionRecord.segment == segment)
+                {
+                    // Collect all the distances for a given ray, defined by the x and z angles
+                    Dictionary<(float, float), List<float>> rayDistances = new();
+                    // for each frame in all the frames collected in a gesture
+                    foreach (var frame in interactionRecord.records)
+                    {
+                        // for each ray in a given frame
+                        foreach (var ray in frame)
+                        {
+                            // if a list hasn't been created for a ray
+                            // i.e. this is the first time the ray is interacting
+                            // in this gesture
+                            if (!rayDistances.ContainsKey((ray.angleX, ray.angleZ)))
+                            {
+                                rayDistances[(ray.angleX, ray.angleZ)] = new List<float>();
+                            }
+                            // add the ray to the list
+                            rayDistances[(ray.angleX, ray.angleZ)].Add(ray.distance);
+                        }
+                    }
+                    int frameCountForMinRayInteractionsThreshold = (int)(minRayInteractionsThreshold * minRayInteractionsThreshold);
```

Comment:
I am not sure what is happening here, the square of `minRayInteractionsThreshold` will always be less that 0, making `frameCountForMinRayInteractionsThreshold` 0? or am I missing something?

Reply:
I did an oopsie, it was supposed to be times the `interactionRecord.records.Count()`